apps/src/lib/levelbuilder/code-docs-editor/ProgrammingExpressionEditor.jsx

```diff
@@ -0,0 +1,134 @@
+import PropTypes from 'prop-types';
+import React, {useState} from 'react';
+import TextareaWithMarkdownPreview from '@cdo/apps/lib/levelbuilder/TextareaWithMarkdownPreview';
+import SaveBar from '@cdo/apps/lib/levelbuilder/SaveBar';
+import {navigateToHref} from '@cdo/apps/utils';
+import $ from 'jquery';
+import color from '@cdo/apps/util/color';
+
+function useProgrammingExpression(initialProgrammingExpression) {
+  const [programmingExpression, setProgrammingExpression] = useState(
+    initialProgrammingExpression
+  );
+
+  function updateProgrammingExpression(key, value) {
+    setProgrammingExpression({...programmingExpression, [key]: value});
+  }
+
+  return [programmingExpression, updateProgrammingExpression];
+}
+
+export default function ProgrammingExpressionEditor({
+  initialProgrammingExpression
+}) {
+  // We don't want to update id or key
+  const {
+    id,
+    key,
+    ...remainingProgrammingExpression
+  } = initialProgrammingExpression;
+  const [
+    programmingExpression,
+    updateProgrammingExpression
+  ] = useProgrammingExpression(remainingProgrammingExpression);
+  const [isSaving, setIsSaving] = useState(false);
+  const [lastUpdated, setLastUpdated] = useState(null);
```

Comment:
i think you mentioned this in a previous comment, but i can't find it -- do we not store a lastUpdated timestamp on the programming expression (in other words: should this always instantiate to `null`)?

Reply:
I've been resolving the comments -- sorry I resolved that one too early! But yes, it should always be instantiated to `null` as we haven't saved it on this page load yet.

`LessonEditor` does the same thing [here](https://github.com/code-dot-org/code-dot-org/blob/56e7cd03dfa43d1f751f1a64247925e8670805f9/apps/src/lib/levelbuilder/lesson-editor/LessonEditor.jsx#L55) for this component, so I wanted to keep that behavior the same.

Reply:
no problem! i thought that was the case, but wanted to double-check